src/pymor/discretizers/builtin/cg.py

```diff
@@ -1178,11 +1203,31 @@ def discretize_stationary_cg(analytical_problem, diameter=None, domain_discretiz
     else:
         output_functional = None
 
+    # assemble additional product
+    if mu_energy_product:
+        if preassemble:
+            # mu_energy_product is the |Parameter| with which we build the energy product (s.a.)
+            eL = eL.assemble(mu_energy_product)
+        else:
+            from pymor.operators.constructions import FixedParameterOperator
+            eL = FixedParameterOperator(eL, mu=mu_energy_product)
+        if p.diffusion is not None:
+            scalar_diffusion = len(p.diffusion.shape_range) == 0 \
+                           or (len(p.diffusion.shape_range) == 1 and p.diffusion.shape_range[0] == 1)
```

Comment:
However, this can never happen. Either, the values of `diffusion` are scalar, or they have to of shape `(grid.dim, grid.dim)`.